src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -82,11 +82,16 @@ public function __construct(bool $magicCall = false, bool $throwExceptionOnInval
      */
     public function getValue($objectOrArray, $propertyPath)
     {
-        $propertyPath = $this->getPropertyPath($propertyPath);
-
         $zval = [
             self::VALUE => $objectOrArray,
         ];
+
+        if (\is_object($objectOrArray) && false === strpos((string) $propertyPath, '.') && false === strpos((string) $propertyPath, '[')) {
```

Comment:
```suggestion
        if (\is_object($objectOrArray) && false === strpbrk((string) $propertyPath, '.[')) {
```

Reply:
Updated, though I wonder if using `strcspn()` and comparing that with `strlen()` would be better in that it doesn't required to copy a string.

Reply:
According to https://blackfire.io/profiles/compare/f5b0113a-fcae-41cc-9b83-3bdd48c8bcf5/graph using `strpbrk ` is faster than `strcspn` + `strlen`.